queue/Queue.java

```diff
@@ -0,0 +1,130 @@
+import java.util.NoSuchElementException;
+
+/**
+ * Implements https://docs.oracle.com/javase/8/docs/api/java/util/Queue.html
+ */
+public class Queue<T> {
+    private int total;
+    // Linked Lists used to store the data
+    private Node first, last;
+
+    private class Node {
+        private T data;
+        private Node next;
+
+        public Node(T data){
+            this.data = data;
+        }
+    }
+
+    /**
+     * Inserts the specified element into this queue if it is possible to do so immediately without violating capacity
+     * restrictions, returning true upon success and throwing an IllegalStateException if no space is currently available.
+     * @param element the element to add
+     * @return  true upon success, throwing an IllegalStateException if no space is currently available.
+     */
+    public boolean add(T element) {
+        Node newNode = new Node(element);
+
+        Node current = last;
+        last = newNode;
+
+        if (total == 0){
+            first = last;
+        } else {
+            current.next = last;
+        }
+        total++;
+
+        return true;
+    }
+
+    /**
+     * Inserts the specified element into this queue if it is possible to do so immediately without violating capacity
+     * restrictions.
+     * @param element the element to add
+     * @return true if the element was added to this queue, else false
+     */
+    public boolean offer(T element) {
+        try{
+            return add(element);
+        } catch (Exception ex){
+            //failed to add, return false
+            return false;
+        }
+    }
+
+    /**
+     * Retrieves and removes the head of this queue.
+     * This method differs from poll only in that it throws an exception if this queue is empty.
+     * @return the head of this queue.
+     * @throws NoSuchElementException if this queue is empty
+     */
+    public T remove() throws NoSuchElementException {
+        if (total == 0) throw new NoSuchElementException();
+        T element = first.data;
+        first = first.next;
+        total --;
+        if(total == 0){last = null;}
+        return element;
+    }
+
+    /**
+     * Retrieves and removes the head of this queue, or returns null if this queue is empty.
+     * This method differs from peek only in that it throws an exception if this queue is empty.
+     * @return the head of this queue, or returns null if this queue is empty.
+     */
+    public T poll() {
+        try{
+            return remove();
+        } catch (Exception ex){
```

Comment:
Would be great if you could change `} catch (..){` to `} catch (...) {` so that it stays consistent everywhere.

Reply:
done (in previous catch statement as well)